test/base.test.js

```diff
@@ -1,12 +1,145 @@
+                    apiKey: 'keyXyz'
+                });
+                const base = fakeAirtable.base('app123');
+                expect.assertions(2);
+                expect(fakeAirtable._endpointUrl).toBe('https://api.airtable.com');
+    
+                try {
+                    await base[authFunction]({
+                        username: 'user',
+                        password: 'password'
+                    });
+                } catch (err) {
+                    expect(err.message).toMatch(`Base#${authFunction} cannot be used with api.airtable.com.`
+                    + ' Please configure this Airlock client to use an Airlock endpoint URL.');
+                }
+            });
+
+            // Correctly configured Airlock client
+            const airtable = new Airtable({
+                apiKey: 'airlock',
+                endpointUrl: 'test',
+                requestTimeout: 1234
+            });
+            let base = airtable.base('app123');
+
+            it('requires username and password', async function() {
+                expect.assertions(2);
+                try {
+                    await base[authFunction]({
+                        password: 'password'
+                    });
+                } catch (err) {
+                    expect(err.message).toMatch(`Missing parameter 'username' required for Base#${authFunction}`);
+                }
+
+                try {
+                    await base[authFunction]({
+                        username: 'user'
+                    });
+                } catch (err) {
+                    expect(err.message).toMatch(`Missing parameter 'password' required for Base#${authFunction}`);
+                }
+            });
+            it('makes authentication requests with the right options', async function() {
+                base = airtable.base('app123');
+
+                base[authFunction]({
+                    username: 'user',
+                    password: 'password'
+                });
+                expect(request).toHaveBeenCalledTimes(1);
+                expect(request).toHaveBeenCalledWith({
+                    method: 'POST',
+                    url: `test/v0/app123/${authFunction}?`,
+                    json: true,
+                    headers: {
+                        authorization: 'Bearer airlock',
+                        'x-api-version': '0.1.0',
+                        'x-airtable-application-id': 'app123',
+                        'x-airtable-user-agent': 'Airtable.js/' + version
+                    },
+                    agentOptions: {
+                        rejectUnauthorized: false
+                    },
+                    body: {
+                        username: 'user',
+                        password: 'password'
+                    },
+                    timeout: 1234
+                }, expect.any(Function));
+            });
+        });
+    });
+
+    describe('#login+logout', function() {
```

Comment:
Can we describe this as `authentication (#login, #logout)`? Also, this test seems to be failing 🤔